module_utils/network_lsr/argument_validator.py

```diff
@@ -2639,7 +2639,13 @@ def _parse_route_tables_mapping(cls, file_content, mapping):
         # It is thus similar to rtnl_rttable_a2n(), from here:
         # https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/lib/rt_names.c?id=11e41a635cfab54e8e02fbff2a03715467e77ae9#n447
         regex = re.compile(
-            b"^\\s*(0x[0-9a-fA-F]+|[0-9]+)\\s+([a-zA-Z0-9_.-]+)(\\s*|\\s+#.*)$"
+            b"^\\s*"  # optional leading whitespace
+            b"(0x[0-9a-fA-F]+|[0-9]+)"  # table ID in hex or decimal (non negative)
+            b"\\s+"  # whitespace
+            b"("  # make a pattern group for the table name
+            + cls._ROUTE_TABLE_NAME_PATTERN.encode("ascii")
+            + b")"  # close pattern group
+            b"(\\s*|\\s+#.*)$"  # trailing whitespace or comment
```

Comment:
Thanks for writting the detailed comment, this is fabulous and much easier to recognize the regex pattern.

Reply:
Thank you.